upload_vehicle_data.py

```diff
@@ -0,0 +1,830 @@
+    command = f'rm {file_path}'
+    run_ssh_command(logger, remote_user, remote_ip, command)
+    logger.debug(f'Deleted file: {file_path}')
+
+
+def find_metadata_file(logger, remote_user, remote_ip, remote_directory):
+    """Find the metadata.yaml file on the remote machine."""
+    find_cmd = f"find {remote_directory} -name 'metadata.yaml'"
+    try:
+        result = subprocess.run(
+            f'ssh {remote_user}@{remote_ip} ' f"'{find_cmd}'",
+            shell=True,
+            capture_output=True,
+            text=True,
+            check=True,
+        )
+        metadata_files = result.stdout.splitlines()
+        if metadata_files:
+            logger.info(f'Found metadata.yaml file at {metadata_files[0]}.')
+            return metadata_files[0]
+        else:
+            logger.error('metadata.yaml file not found.')
+            return None
+    except subprocess.CalledProcessError as e:
+        logger.error(f'Failed to find metadata.yaml file: {e}')
+        raise
+
+
+def copy_metadata_file(
+    logger,
+    remote_user,
+    remote_ip,
+    metadata_path,
+    cloud_upload_directory,
+    remote_directory,
+):
+    """Copy metadata.yaml file from remote machine to cloud host."""
+    relative_metadata_path = os.path.relpath(
+        metadata_path, start=remote_directory
+    )
+    rsync_cmd = [
+        'rsync',
+        '-av',
+        '--checksum',
+        '--progress',
+        '--stats',
+        f'{remote_user}@{remote_ip}:{metadata_path}',
+        os.path.join(cloud_upload_directory, relative_metadata_path),
+    ]
+    try:
+        logger.info(f'Uploading {relative_metadata_path}.')
+        subprocess.run(rsync_cmd, check=True)
+        logger.debug(f'Copied metadata.yaml to {cloud_upload_directory}.')
+        return True
+    except subprocess.CalledProcessError as e:
+        logger.error(f'Failed to copy metadata.yaml: {e}')
+        return False
+
+
+def read_metadata(logger, metadata_path):
+    """Read the metadata.yaml file and return its contents."""
+    with open(metadata_path) as file:
+        metadata = yaml.safe_load(file)
+    logger.debug(f'Read metadata from {metadata_path}.')
+    return metadata
+
+
+def check_and_create_local_directory(logger, directory_path):
+    """Check if a directory exists on the local machine and create it."""
+    if not os.path.exists(directory_path):
+        try:
+            os.makedirs(directory_path)
+            logger.debug(f'Created local directory: {directory_path}')
+        except OSError as e:
+            logger.error(f'Failed to create directory {directory_path}: {e}')
+            raise
+
+
+def process_directory(
+    logger,
+    remote_user,
+    remote_ip,
+    remote_directory,
+    cloud_upload_directory,
+    config,
+    base_remote_directory,
+    bandwidth_mbps,
+    file_sizes_dict,
+    files_dict,
+    global_rosbag_counter,
+    total_rosbags,
+):
+    """Process each directory."""
+    logger.info(f'Processing {remote_directory}')

```

Comment:
Lets add a white space for every subdirectory we process
```suggestion
    logger.info('')
    logger.info(f'Processing: {remote_directory}')

```

Reply:
Changed.